# Exploracion Bases De Datos

## Cats - Dogs (Audios)
- **Fuente**: [Audio Cats and Dogs - Kaggle](https://www.kaggle.com/datasets/mmoreaux/audio-cats-and-dogs)
- **Tipo De Datos**: Esta es una base de datos **secundaria**, ya que la información fue recopilada por otras personas. Se trata de datos de acceso público que no fueron recolectados específicamente con el propósito de realizar un análisis.
    - Los datos costan de audios de gatos y perros en formato .wav de 16KHz y duracion variable.

A continuacion el calculo de las caracteristicas basicas.


In [12]:
import os

# Ruta de la carpeta
folder_path = "cats_dogs"

# Listas para almacenar los nombres de los archivos
all_files = []
cat_files = []
dog_files = []

# Recorremos todos los archivos de la carpeta
for file in os.listdir(folder_path):
    if os.path.isfile(os.path.join(folder_path, file)):  # verificamos que sea un archivo
        all_files.append(file)
        if "cat" in file.lower():   # buscar 'cat' sin importar mayúsculas
            cat_files.append(file)
        elif "dog" in file.lower(): # buscar 'dog' sin importar mayúsculas
            dog_files.append(file)

# Resultados
print("Cantidad total de archivos:", len(all_files))
print("Cantidad de archivos para 'cat':", len(cat_files))
print("Cantidad de archivos para 'dog':", len(dog_files))


Cantidad total de archivos: 277
Cantidad de archivos para 'cat': 164
Cantidad de archivos para 'dog': 113


- **Caracteristicas Basicas**
    - Cantidad **total** de archivos: 277
    - Cantidad de archivos para **'cat'**: 164
    - Cantidad de archivos para **'dog'**: 113
- **Posibles Aplicaciones**: Clasificación de sonidos, Reconocimiento de animales en ambientes reales, Detección temprana en dispositivos IoT (ej. Alexa)


# Rayos X Neumonia (Imagenes)

- **Fuente**: [Chest X-Ray Images (Pneumonia) - Kaggle](https://www.kaggle.com/code/shyambv/starter-chest-x-ray-images-pneumonia-0f43d8bd-7)
- **Tipo De Datos**: Esta es una base de datos **secundaria**, ya que la información fue recopilada por otras personas. Se trata de datos de acceso público que no fueron recolectados específicamente con el propósito de realizar un análisis.
    - Los datos costan de imagenes de rayos x de personas con y sin neumonia, el formato de las imagenes es jpeg con dimensiones variables.

A continuacion el calculo de las caracteristicas basicas.

In [ ]:
# Rutas de las carpetas
train_path = "chest_xray/train"
normal_path = os.path.join(train_path, "NORMAL")
pneumonia_path = os.path.join(train_path, "PNEUMONIA")

# Función para contar imágenes en una carpeta
def contar_imagenes(carpeta):
    return len([f for f in os.listdir(carpeta) if os.path.isfile(os.path.join(carpeta, f))])

# Contar imágenes
normal_count = contar_imagenes(normal_path)
pneumonia_count = contar_imagenes(pneumonia_path)

# Resultados
print("Cantidad total de imágenes:", normal_count + pneumonia_count)
print("Cantidad de imágenes de personas NORMAL:", normal_count)
print("Cantidad de imágenes de personas con PNEUMONIA:", pneumonia_count)

FileNotFoundError: [WinError 3] The system cannot find the path specified: 'chest_xray/test\\NORMAL'

- **Caracteristicas Basicas**
    - Cantidad total de imágenes: 5216
    - Cantidad de imágenes de personas **NORMAL**: 1341
    - Cantidad de imágenes de personas con **PNEUMONIA**: 3875
- **Posibles Aplicaciones**: Diagnóstico asistido por IA, Detección temprana y triage, Educación médica

# Ventas Shopify (Tabular)